File: 2053-kth-distinct-string-in-an-array/2053-kth-distinct-string-in-an-array.py
class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:
        hash_map = {}
        for i in arr:
            if i in hash_map:
                hash_map[i] = False
            else:
                hash_map[i] = True
        
        count = 1
        print(hash_map)
        for i in hash_map:
            if hash_map[i] and count == k:
                return i
            if hash_map[i]:
                count += 1
                
        return ""

File: 2053-kth-distinct-string-in-an-array/2053-kth-distinct-string-in-an-array.java
class Solution {
    public String kthDistinct(String[] arr, int k) {
        Map<String, Integer> map = new LinkedHashMap<>();
        for (String str: arr) {
            map.put(str, map.getOrDefault(str, 0) + 1);
        }

        for (Map.Entry<String, Integer> entry: map.entrySet()) {
            String str = entry.getKey();
            int frequency = entry.getValue();
            if (frequency == 1) {
                k--;
                if (k == 0) {
                    return str;
                }
            }
        }
        return "";
    }
}